tests/test_url_building.py

```diff
@@ -39,6 +39,30 @@ def test_simple_url_for_getting(simple_app):
         assert response.text == letter
 
 
+def test_simple_url_for_getting_with_duplicate_params(simple_app):
+    kw = dict(arg1=['value1', 'value2'], _anchor='anchor')
+    for letter in string.ascii_letters:
```

Comment:
I'd say `pytest.mark.parameterize` would be good for these tests. It makes sense to do it that way so we can have them run as separate tests instead of all at once.